package/MDAnalysis/coordinates/TRC.py

```diff
@@ -0,0 +1,409 @@
+   The manual can be downloaded here:
+   https://gromos.net/gromos11_pdf_manuals/vol4.pdf
+
+This reader is designed to read the blocks "TIMESTEP", "POSITIONRED" and
+"GENBOX" from the trajectory which covers the standard trajectories of most
+simulations . If a trajectory includes other blocks, a warning is served
+and the blocks are ignored.
+
+.. Note::
+   MDAnalysis requires the blocks to be in the same order for each frame
+   and ignores non-supported blocks.
+
+
+
+Classes
+-------
+
+.. autoclass:: TRCReader
+   :members:
+
+"""
+
+import pathlib
+import errno
+import warnings
+import numpy as np
+
+from . import base
+from .timestep import Timestep
+from ..lib import util
+from ..lib.util import cached, store_init_arguments
+
+import logging
+
+logger = logging.getLogger("MDAnalysis.coordinates.GROMOS11")
+
+
+class TRCReader(base.ReaderBase):
+    """Coordinate reader for the GROMOS11 format"""
+
+    format = "TRC"
+    units = {"time": "ps", "length": "nm"}
+    _Timestep = Timestep
+
+    SUPPORTED_BLOCKS = ["TITLE", "TIMESTEP", "POSITIONRED", "GENBOX"]
+    NOT_SUPPORTED_BLOCKS = [
+        "POSITION",
+        "REFPOSITION",
+        "VELOCITY",
+        "VELOCITYRED",
+        "FREEFORCE",
+        "FREEFORCERED",
+        "CONSFORCE",
+        "CONSFORCERED",
+        "SHAKEFAILPOSITION",
+        "SHAKEFAILPREVPOSITION",
+        "LATTICESHIFTS",
+        "COSDISPLACEMENTS",
+        "STOCHINT",
+        "NHCVARIABLES",
+        "ROTTRANSREFPOS",
+        "PERTDATA",
+        "DISRESEXPAVE",
+        "JVALUERESEXPAVE",
+        "JVALUERESEPS",
+        "JVALUEPERSCALE",
+        "ORDERPARAMRESEXPAVE",
+        "XRAYRESEXPAVE",
+        "LEUSBIAS",
+        "LEUSBIASBAS",
+        "ENERGY03",
+        "VOLUMEPRESSURE03",
+        "FREEENERGYDERIVS03",
+        "BFACTOR",
+        "AEDSS",
+    ]
+    
+    @store_init_arguments
+    def __init__(self, filename, convert_units=True, **kwargs):
+        super(TRCReader, self).__init__(filename, **kwargs)
+
+        # GROMOS11 trajectories are usually either *.trc or *.trc.gz.
+        # (trj suffix can come up when trajectory obtained from clustering)
+        ext = pathlib.Path(self.filename).suffix
+        if (ext[1:] == "trc") or (ext[1:] == "trj"):
+            self.compression = None
+        else:
+            self.compression = ext[1:]
+
+        self.trcfile = util.anyopen(self.filename)
+        self.convert_units = convert_units
+
+        # Read and calculate some information about the trajectory
+        self.traj_properties = self._read_traj_properties()
+
+        self._cache = {}
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+
+        self._reopen()
+        self.ts.dt = self.traj_properties["dt"]
+
+        self._read_frame(0)
+
+    @property
+    @cached("n_atoms")
+    def n_atoms(self):
+        """The number of atoms in one frame."""
+        return self._read_atom_count()
+
+    def _read_atom_count(self):
```

Comment:
Iirc the `cached` decorator already guards against this code getting ran twice, (it's something like, if the data exists, return it without entering the function body), so splitting the method into a private section probably isn't necessary.  Not harmful though.

Reply:
I agree with you that it is nice to merge these associated functions.
Done.